app/resources/api/v2/barcode_printer_resource.rb

```diff
@@ -33,11 +33,11 @@ class BarcodePrinterResource < BaseResource
 
       # @!attribute [r] print_service
       #   @return [String] the service this printer can be instructed to print from.
-      attribute :print_service, readonly: true
+      attribute :print_service, write_once: true
```

Comment:
Im not sure if these should be write_once since this is an immutable resource and I don't think printers are created through the API (to my knowledge).

Reply:
I'm happy to change this back to `readonly` if you believe that's what we need.  Bear in mind that before I changed this, `readonly` actually meant `write_once` (bad naming in the past).  So by changing this to `write_once` this is a "no actual change made" amendment.  Anything I've left as `readonly` is now LESS permissive than before.

Reply:
Okay, happy to go with your decision. Does write_once still do anything when the resource is immutable?

Reply:
Technically yes, but `immutable` makes it impossible to hit the endpoint with a method that modifies resource.  When you put `immutable` in a resource, the `jsonapi_resources` entry in `routes.rb` only sets up a GET endpoint and you cannot submit a `POST` or `PATCH` method request.